### PR TITLE
Add OCP support to helm chart

### PR DESCRIPTION
**Pull request by binguo-casa**
This setup has been tested with OCP 4.6 and 4.7, and is based on these previous discussions:
- okd 4.5: https://github.com/longhorn/longhorn/issues/1831#issuecomment-702690613
- okd 4.6: https://github.com/longhorn/longhorn/issues/1831#issuecomment-765884631
- oauth-proxy: https://github.com/openshift/oauth-proxy/blob/master/contrib/sidecar.yaml

Main changes and tasks for OCP are:
- security setup for namespace longhorn-system
- option to use separate disk in /var/mnt/longhorn
- exposing longhorn ui via oauth-proxy
- adding finalizers for mount propagation
- machineconfig file to start iscsid
- mahcineconfig file to mount /var/mnt/longhorn

### CODE REVIEW

**Comment by innobead** [comment]
@binguo-casa Thanks for the contribution! Please help resolve the DCO check failure first. ref: https://github.com/longhorn/longhorn/pull/2909/checks?check_run_id=3383674759

**Comment by innobead** [comment]
@khushboo-rancher let's check this after 1.2.0

**Comment by binguo-casa** [comment]
Missing instancemanagers/finalizers, will figure out how to update:

```--- a/longhorn/helm/1.2.0/longhorn/templates/clusterrole.yaml
+++ b/longhorn/helm/1.2.0/longhorn/templates/clusterrole.yaml
@@ -38,7 +38,7 @@ rules:
   resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
               "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
   {{- if .Values.openshift.finalizers }}
-              "engineimages/finalizers", "nodes/finalizers",
+              "engineimages/finalizers", "nodes/finalizers", "instancemanagers/finalizers",
   {{- end }}
               "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
               "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
```

File: chart/values.yaml [hunk 1]
```diff
@@ -41,7 +41,11 @@ image:
     snapshotter:
       repository: k8s.gcr.io/sig-storage/csi-snapshotter
       tag: v3.0.3
-  pullPolicy: IfNotPresent
+  openshift:
+    oauthProxy:
+      repository: openshift/oauth-proxy
+      tag: latest
+deploy/release-images.txt  pullPolicy: IfNotPresent
```
Comment (jenting) [inline chart/values.yaml:48]: Is the format correct?

**Review by jenting — COMMENTED** [review]
I did not have the env to test this chart enhancement for OCP.
So I review it from the chart manifest.

**Comment by innobead** [comment]
@binguo-casa could you help resolve the conflicts? thanks.

**Comment by ArthurVardevanyan** [comment]
Form my Testing ```kind: MachineConfig``` is no longer required.

**Comment by innobead** [comment]
This seems outdated. Let's do https://github.com/longhorn/longhorn/pull/5004 instead. 

@binguo-casa feel free to be involved in the review of 5004. Thanks for the contribution.

**Pull request closed without merging**